src/common/dout.h

```diff
@@ -171,10 +171,25 @@ struct is_dynamic<dynamic_marker_t<T>> : public std::true_type {};
   } while (0)
 #endif	// WITH_SEASTAR
 
+// like dout_impl, but first occurrence will be logged at level 0
+#define dout_impl_once(cct, sub, v)					\
```

Comment:
Pity that wouldn't work.

I'm not sure what you mean with the clog though. We're being handed an extraction, and I don't think we can "tee" that into multiple destinations. The clog looks like it might not be appropriate for this either? In any case, that sounds like a much more involved project than what I had proposed.

At this point, I have a patch that I hope will fix this on the kernel side. In the meantime I may just close this out and we can reconsider it if it turns out that we need to investigate further.

Reply:
My suggest was not to "tee" the log message to multiple destinations. Make a brand new message that's only done once. 

I think the key thing here is that this situation **does** indicate a bug of some sort, perhaps just an old client (without your kclient fix). The admin should be spurred in ways beyond a more verbose debug message in the log. Perhaps we could have a flag set on the session for the client which indicates it is sending cap flushes that are getting dropped by the MDS. Then the beacons sent to the mon could include a health warning for that. (See `Beacon::notify_health`.)